Docs/developer_guide/mrml_overview.md

```diff
@@ -100,14 +100,299 @@ The following methods on the MRML scene are used to manage Undo/Redo stacks:
+* Inherits from `qSlicerNodeWriter`.
+* See the reader for information on defining and constructing Qt style classes.
+* Override `QStringList extensions(vtkObject* object) const` to provide file extensions that can be written to.
+  * File extensions may be different, but don’t have to be, for different data nodes that in the same hierarchy (e.g. Markups Curve and Plane could reasonably require different file extensions, but they don’t).
+* Override `bool write(const qSlicerIO::IOProperties& properties)` to do the actual writing (by way of a storage node, of course).
+
+#### The subject hierarchy plugin
+
+A convenient module in 3DSlicer is the Data module. It brings all the different data types together under one roof and offers operations such as cloning, deleting, and renaming nodes that work regardless of the node type. The Data module uses the Subject Hierarchy, which is what we need to plug into so our new node type can be seen in and modified by the Data module.
+
+Files:
+
+* \<Extension>/\<Module>/SubjectHierarchyPlugins/qSlicerSubjectHierarchy\<MyCustomType>Plugin.h
+* \<Extension>/\<Module>/SubjectHierarchyPlugins/qSlicerSubjectHierarchy\<MyCustomType>Plugin.cxx
+
+Key Points:
+
+* Naming convention for class: `qSlicerSubjectHierarchy<MyCustomType>Plugin`
+* Inherits from `qSlicerSubjectHierarchyAbstractPlugin`.
+* See the reader for information on defining and constructing Qt style classes.
+* Override `double canAddNodeToSubjectHierarchy(vtkMRMLNode* node, vtkIdType parentItemID=vtkMRMLSubjectHierarchyNode::INVALID_ITEM_ID) const`.
+  * This method is used to determine if a data node can be placed in the hierarchy using this plugin.
+* Override `double canOwnSubjectHierarchyItem(vtkIdType itemID) const` to say if this plugin can own a particular subject hierarchy item.
+* Override `const QString roleForPlugin() const` to give the plugin’s role (most often meaning the data type the plugin can handle, e.g. Markup).
+* Override `QIcon icon(vtkIdType itemID)` and `QIcon visibilityIcon(int visible)` to set icons for your node type.
+* Override `QString tooltip(vtkIdType itemID) const` to set a tool tip for your node type.
+* Override the following to determine what happens when a user gets/sets the node color through the subject hierarchy:
+  * `void setDisplayColor(vtkIdType itemID, QColor color, QMap<int, QVariant> terminologyMetaData)`
+  * `QColor getDisplayColor(vtkIdType itemID, QMap<int, QVariant> &terminologyMetaData) const`
+
+#### The module (aka putting it all together)
+
+If you have used 3DSlicer for any length of time, you have probably noticed that for each type of node (or set of types as in something like markups) there is a dedicated module that is used solely for interacting with the single node type (or set of types). Examples would be the Models, Volumes, and Markups modules. These modules are useful from a user perspective and also necessary to get your new node registered everywhere it needs to be.
```

Comment:
We try to always use "3D Slicer" not "3DSlicer"

Reply:
Update to use "3D Slicer"